src/server.py

```diff
@@ -2,15 +2,28 @@
 import json
 import os
 import urlparse
+import subprocess
+import glob
+import warnings
 from BaseHTTPServer import BaseHTTPRequestHandler, HTTPServer
 
 from rasa_nlu.util import update_config
 
 
 def create_interpreter(config):
     model_dir = config.get("server_model_dir")
-    metadata, backend = None
-    if model_dir is not None:
+    metadata, backend = None, None
+
+    if (model_dir is not None):
+        # download model from S3 if needed
+        if (not os.path.isdir(model_dir)):
```

Comment:
I find this behavior a little odd. So as soon as there is no local model it will get downloaded from S3? Might lead to strange errors (e.g. I want to use a local file, but put in the wrong name and might not get an error because the model gets downloaded from S3).

Can you explain the use case here? E.g. what is the most common use case for the S3 connection